### DISCUSSION TITLE
Auto Save

### DISCUSSION THREAD

**Discussion by karimStekelenburg** [outline/outline#2203] [Ideas]
Before I start whining, let me be clear. I've been searching for a wiki / knowledge base solution for a while now for my company, and Outline is by far the best one I've encountered. We've recently switched from Wiki.JS to Outline and all of us fell in love with it. However, recently I ran into some trouble.

I was editing a page for about an hour and accidentally closed the tab. When I reopened the page I expected my draft to still be there. However, this wasn't the case and all my work was lost.

Now hear me out. I understand I shouldn't work on a page for an hour without manually saving my work now and then, but I'm an idiot who doesn't always think about those things.

I'd love to see a feature that auto-saves my edits without me having to manually click `save draft` or `done editing` in case I fudge up. I don't need to be able to recover every single character I typed, but I think an auto-save every minute or so is a useful feature.

I'm currently copy-pasting everything from-and-to a local markdown editor whenever I'm editing a page, only to avoid this issue. However, I love the Outline editor and would like to use it as my primary editor.

Keep up the work! I usually don't like subscriptions, but Outline is definitely worth it ♥️

**Comment by tommoor** [comment]
Hey Karim, this is really strange as Outline _has_ autosave and has had for many years, the save is debounced after several seconds of stopping typing so if working for an hour it should have been saved many many times.

This makes me wonder if the saves were silently failing and if so what might have caused that to occur. If you're able to email support hello@getoutline.com with some details on the doc we can check the logs to get some more clues.

**Reply by karimStekelenburg** [reply]
Hi Tom, thanks for the quick reply. Good to hear it's already implemented, as it's kind of a fundimental feature. I'll email a more detailed report tomorrow. Is the draft stored server-side or on the client? If the latter is the case, is there some stuff I can checkout (e.g. local storage) and attach to the email?

Thanks again :-)

**Reply by tommoor** [reply]
> Is the draft stored server-side or on the client?

On the server side
